build/llvm.yml

```diff
@@ -0,0 +1,231 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+trigger: none
+
+variables:
+  CCACHE_DIR: $(Pipeline.Workspace)/ccache
+  CMAKE_INSTALL_PREFIX: $(Pipeline.Workspace)/LLVM
+  INSTALL_LLVM_PACKAGE: true
+  SCCACHE_DIR: $(Pipeline.Workspace)\sccache
+  AQ_LLVM_PACKAGE_GIT_VERSION: ""
+  ENABLE_COMPILER: false
+  ENABLE_VSIX: false
+  ENABLE_LLVM_BUILDS: true
+  AQ_DOWNLOAD_LLVM: false
+
+jobs:
+- job: GetLlvmTag
+  pool:
+    vmImage: 'ubuntu-20.04'
+  steps:
+  - checkout: self
+  - pwsh:  Write-Host "##vso[task.setvariable variable=AQ_LLVM_PACKAGE_GIT_VERSION;isoutput=true]$(Get-Content ./llvm.tag.txt)"
+    workingDirectory: $(System.DefaultWorkingDirectory)/llvm-build
+    displayName: "Get LLVM Tag"
+    name: Tag
+
+- job: LLVMLinuxBuild
+  timeoutInMinutes: 300
```

Comment:
5-hour time-out? If the [pipeline runs for 20 minutes](https://github.com/microsoft/qsharp-compiler/pull/1299#issuecomment-1009438656), then 5-hour-long time-out seems redundantly long. I would consider 40 - 60 minutes.

Reply:
This one is a bit tricky... the pipeline runs for 20 minutes if the cache is fully populated, but if not, or if the LLVM version has been updated to one where a cached build does not exist, the build takes about 1.5 hours. In testing, I believe @idavis saw builds that took upwards of 3 hours if additional features were enabled. So under normal circumstances, 20 minutes is the expected build time, but when intentionally doing an update or running after cache expiration, then it could be longer.

Incidentally, this is why the intent is to not build this pipeline unless manually invoked; regular PRs should not be producing any changes to the binary dependencies we have on LLVM.